roles/ffh.mesh_wireguard/tasks/netlink.yml

```diff
@@ -0,0 +1,49 @@
+---
+
+- name: ensure wireguard config directory
+  file:
+    path: /etc/wireguard
+    state: directory
+
+- name: place netlink config
+  template:
+    src: netlink_cfg.json.j2
+    dest: /etc/wireguard/netlink_cfg.json
+
+- name: ensure wireguard srv directory
+  file:
+    path: /srv/wireguard
+    state: directory
+
+- name: install pip3
```

Comment:
Install all dependencies at the beginning

Reply:
Why? this way the dependency is directly before is neccessity

Reply:
To me it's more expected behaviour that dependencies are the first thing that are being checked before touching anything. Dependencies are **pre**requisites.

Reply:
Indeed pre the task their needed for. Which they are.
Seriously, may someone else settle this? We obviously both are hurt in our ocd.

Reply:
@AiyionPrime I've just run into an issue with your role as ansible tries to install pyroute2 using pip2 while it installed pip3. The frustration of the user is definitely higher if your role does something and then in the middle aborts because of missing dependencies. One of the most frustrating points when building Gluon is if it fails because of wrong site config parameters right before being finished... It's the same here. You are against having the dependencies at the beginning of the actual role because of your aesthetic pleasure as a developer, but from the user perspective it looks wrong and they won't complain about your code I guess.

Reply:
"You've got to start with the customer experience (...)":
https://youtube.com/watch?v=oeqPrUmVz-o&t=1m30s

Reply:
As far as I know somebody already merged this PR.

Reply:
At the moment it's still just a philosophical discussion. It's no functional issue. So more of a religious, metaphysical mindset thing...